### DISCUSSION TITLE
Peloton API ?

### DISCUSSION THREAD

**Discussion by GregChung-IBFL** [philosowaffle/peloton-to-garmin#90] [Q&A]
I don't know about the terms of use for the Peloton API.  I do have some questions, in case anyone is familiar:

How long is the session valid for?

Would it be abusive to, for example, pull down 500 activities in a short time?  Basically, if I wanted to pull my entire history into my own database, I suppose I could add some sleeps to spread it out rather than spamming the API.  Technically, writing our own apps might violate the terms of use agreement.

Also wondering if there really isn't a method to pull activities by timestamp, as opposed to last N workouts?

**Comment by philosowaffle** [comment]
> How long is the session valid for?

Not sure, I haven't ever been patient enough to test and find out.  Maybe someone else here knows.

> Terms of use?

There are no terms of use documented that I'm aware of.  Though generally speaking I consider using this tool and the api under the `Use at your own risk` umbrella.  I experimented with calling Peloton once every hour to poll for new workouts, made it about 48hrs before they reset my password because they had detected suspicious activity on my account.  I do know that @daytonbrown successfully backfilled 150 workouts.  Would be curious to hear about other people's experience with this.

> Download by timestamp?

This can be done.  I personally just haven't gotten around to it.  Unfortunately, the [API only allows us to query by N number of workouts](https://app.swaggerhub.com/apis/DovOps/peloton-unofficial-api/0.2.9#/users/get_api_user__userId__workouts), so fetching back to a certain date requires a bit of looping and code to accomplish.

**Reply by GregChung-IBFL** [reply]
Dang.  I'll try writing something custom to just download my history over time.  It'd be like scrolling back and clicking each workout one by one, I suppose.
What I would like to do (admittedly not specifically related to this project) is to download my history and make is searchable.  For example, I think it's silly that the Peloton interface can tell me I've collected "Gratitude Week 2020" special event badge 9 times, but there's no way to see those activities.  That level of detail is only seen by drilling down into every workout.  Another thing is, perhaps I'd like to query the number of times I've taken classes by each instructor.  That's another thing which the real app ought to display.  If I collect the full workout details into my own database, it'd be easy to set up queries like those.
